.github/workflows/test.yml

```diff
@@ -58,3 +58,7 @@ jobs:
     - name: run regression tests
       run: |
        make test
+    - name: code coverage
+      run: |
+       PYTHON="coverage run" make test_unittest
+       make coverage_report
```

Comment:
Are there plans to visually represent per-line coverage using, say, coveralls.io? It should be easy to implement and will be a way more useful comparing to just numbers.

Reply:
Added command to execute coveralls tool in GH Actions, it is required to pass an auth token in an environment variable. Could you set it there?

https://coveralls-python.readthedocs.io/en/latest/usage/configuration.html

Reply:
[GITHUB_TOKEN](https://docs.github.com/en/actions/reference/authentication-in-a-workflow) is always there. I added `COVERALLS_TOKEN` into secrets just in case.